### DISCUSSION TITLE
Error using offset terms in Form() and Persist() with new version of tergm?

### DISCUSSION THREAD

**Discussion by JeffreyAlanSmith** [statnet/tergm#113] [Q&A]
Hi folks, I have a quick question about specifying offset terms as part of the Form or Persist equations within tergm. Previously this was possible using the offset() function, but that leads to errors with the newest version of the package. Take the example below: 

library(tergm)
data(samplk)
samp <- list(samplk1, samplk2, samplk3)
tergm.fit <- tergm(samp ~ Form(~ edges + offset(nodematch("group"))) + 
                          Persist(~ edges + offset(nodematch("group"))), 
                          offset.coef = c(1, 1), estimate = "CMLE", times=0:2)

Error in `ergm_Init_abort()`:
! In term ‘N’ in package ‘ergm.multi’ (called from term ‘Form’ in package ‘tergm’): The ERGM ‘formula=’ argument of an ‘N()’ operator may not have offsets. See ‘ergmTerm?N’ section on fixing parameters for details.
Run `rlang::last_trace()` to see where the error occurred.

I am not sure if this is a bug or by design but how would I specify a formula with offset terms in the current version of the package? I didn't see any examples in the help files, but maybe I missed them? 

Much appreciated, 

Jeffrey

**Comment by mbojan** [comment]
@JeffreyAlanSmith thanks for that inquiry. For that scenario I think you'd need to use the `offset=` argument to `Form()` and `Persist()`, but I'll pass the baton to @krivit to corroborate if it is the case and how to do it exactly.

**Comment by krivit** [comment]
> I am not sure if this is a bug or by design but how would I specify a formula with offset terms in the current version of the package? I didn't see any examples in the help files, but maybe I missed them?

Please see the section in `ergmTerm?N` documentation on specifying offsets and fixing parameters. It also applies to operators that use it, such as `Form()` and `Persist()`.

**Comment by JeffreyAlanSmith** [comment]
Thanks Michal (@mbojan) and Pavel (@krivit). I'll see if I can find a simple way to specify this.

**Comment by mbojan** [comment]
@JeffreyAlanSmith , would that be the specification along your intentions?

```r
fit <- tergm(
  samp ~ offset(Form(~ nodematch("group"))) + Form(~ edges) +
    offset(Persist(~ nodematch("group"))) + Persist(~ edges),
  estimate = "CMLE", 
  offset.coef = c(1, 1),
  times=0:2
)

summary(fit)
```

```
Call:
tergm(formula = samp ~ offset(Form(~nodematch("group"))) + Form(~edges) + 
    offset(Persist(~nodematch("group"))) + Persist(~edges), offset.coef = c(1, 
    1), estimate = "CMLE", times = 0:2)

Conditional Maximum Likelihood Results:

                                   Estimate Std. Error MCMC % z value Pr(>|z|)    
offset(Form(1)~nodematch.group)      1.0000     0.0000      0     Inf   <1e-04 ***
Form(1)~edges                       -2.7231     0.1701      0 -16.011   <1e-04 ***
offset(Persist(1)~nodematch.group)   1.0000     0.0000      0     Inf   <1e-04 ***
Persist(1)~edges                     0.1375     0.2065      0   0.666    0.506    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

     Null Deviance: 848.4  on 612  degrees of freedom
 Residual Deviance: 373.8  on 608  degrees of freedom
 
AIC: 377.8  BIC: 386.6  (Smaller is better. MC Std. Err. = 0)

 The following terms are fixed by offset and are not estimated:
  offset(Form(1)~nodematch.group) offset(Persist(1)~nodematch.group) 
```


PS. Transformed that thread into a Q&A Discussion.
